awx/lib/awx_display_callback/events.py

```diff
@@ -31,8 +31,8 @@
 try:
     import memcache
 except ImportError:
-    raise ImportError('python-memcached is missing; {}bin/pip install python-memcached'.format(
-        os.environ['VIRTUAL_ENV']
+    raise ImportError('python-memcached is missing; {}/bin/pip install python-memcached'.format(
```

Comment:
While I fully agree about trailing slashes, i don't understand why not use pathlib or at least os.path.join ? It seems like the code deals with that more than it should in general.

Reply:
I agree this would be better as `os.path.join(os.environ['VIRTUAL_ENV'], 'bin', 'pip')`